### ISSUE TITLE
Freeze while assigning data from numpy array when the data is a string

### ISSUE THREAD

**Issue by albertogomcas** [spyder-ide/spyder#2983]
The following script is a minimum sample that freezes Spyder (2.3.8, with Python 2.7.11 64bits, Qt 4.8.7, PyQt4 (API v2) 4.11.4 on Windows) on my computer.

    import numpy as np

    np.savez("killertext.npz", mytext = u'abcd')
    data = np.load("killertext.npz")

    print(data["mytext"]) #ok
    text1 = str(data["mytext"]) #ok
    text2 = data["mytext"] #this assignment freezes spyder !

Additional info:
The same script runs fine in the standalone python interpreter.
It does not matter if the string is unicode or not

**Comment by ccordoba12** [comment]
Sorry but I can't reproduce this error. I tried with Numpy 1.9 and 1.10 and Spyder doesn't show freezes.

@dmenin, @wolfobert, just to let you know about this. You all gave me simple and easy ways to reproduce this problem. I tried them all with no avail :-s

**Comment by dmenin** [comment]
Hey @ccordoba12 

I can reproduce the error with:
numpy '1.10.4'
spyder 2.3.8

Python 2.7.11 |Anaconda 2.5.0 (64-bit)| (default, Jan 29 2016, 14:26:21) [MSC v.1500 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 4.0.3 -- An enhanced Interactive Python.

Diego

**Comment by ccordoba12** [comment]
@dmenin, the problem is if I can't reproduce this problem, I can't fix it :-)

To help me, please start Spyder from a terminal (`cmd.exe`) with this command:

    spyder --show-console

then generate the freeze by creating a numpy array with strings.

After that, please go back to the terminal and press `Ctrl+C` in there. That should unfreeze Spyder, but most importantly it should generate the error message that's causing it in Spyder *Internal Console*.

If you're unable to see that message, please go to the menu

    View > Panes > Internal Console

to see it.

Finally, please copy that error message and post it here. With that info I can fix this problem :-)

**Comment by dmenin** [comment]
Hey Carlos, this is all I get, hope this helps:

```
C:\>spyder --show-console 
forrtl: error (200): program aborting due to control-C event
Image              PC                Routine            Line        Source
libifcoremd.dll    000007FEDBFD4314  Unknown               Unknown  Unknown
kernel32.dll       0000000077314B93  Unknown               Unknown  Unknown
kernel32.dll       00000000772D59ED  Unknown               Unknown  Unknown
ntdll.dll          000000007750B371  Unknown               Unknown  Unknown
QObject::killTimers: timers cannot be stopped from another thread
QObject::killTimers: timers cannot be stopped from another thread
```

**Comment by ccordoba12** [comment]
So Spyder just closes and it's not unfrozen?

**Comment by dmenin** [comment]
yes, should have mentioned that.

**Comment by ccordoba12** [comment]
Bad news. I think the function that is generating the freeze is `value_to_display`, located here

https://github.com/spyder-ide/spyder/blob/2.3/spyderlib/widgets/dicteditorutils.py#L172

If you want to fix this, please try to play with it and see if you can avoid the freeze.

**Comment by ccordoba12** [comment]
@dmenin, I think I found the error. Do you have the **Show arrays min/max** option activated in the Variable Explorer? I mean, this option?

![seleccion_002](https://cloud.githubusercontent.com/assets/365293/13928947/d38398ec-ef65-11e5-9d7b-30e2a03f0e8e.png)

With that option activated I can generate a freeze with NumPy arrays with strings on them (and it's not only caused on Windows :-)

**Comment by dmenin** [comment]
yes, you are right, if I uncheck the option I dont' get the error
Good catch.

**Comment by ccordoba12** [comment]
Thanks for the confirmation @dmenin. This error will be fixed in 2.3.9 :-)

**Comment by dmenin** [comment]
thank you :)
